### PR TITLE
Use std::filesystem::path instead of string for file paths

### PR DESCRIPTION
**Pull request by dschuff**

### CODE REVIEW

File: src/support/file.h [hunk 1]
```diff
@@ -69,11 +70,8 @@ class Output {
   std::ostream out;
 };
 
-// Copies a file to another file
-void copy_file(std::string input, std::string output);
-
-// Retusn the size of a file
-size_t file_size(std::string filename);
+// Copies a file to another file, overwriting if the fiel exists
```
Comment (kripken) [inline src/support/file.h:73]:
```suggestion
// Copies a file to another file, overwriting if the target exists
```

**Review by kripken — COMMENTED** [review]
Looks like a good direction to me to do this gradually!

CI errors look a little worrying though in terms of cross-platform support for this API?

**Comment by dschuff** [comment]
Switching to draft, Going to try a couple of things to maybe avoid compatibility issues...

**Comment by brson** [comment]
This looks like a good first step toward supporting unicode paths on windows. Thanks.

**Comment by dschuff** [comment]
Closing this in favor of #5671

**Pull request closed without merging**